src/oss-reproducible/Helpers/Helpers.cs

```diff
@@ -0,0 +1,441 @@
+    }
+
+    public class Helpers
+    {
+        
+        protected static readonly NLog.Logger Logger = NLog.LogManager.GetCurrentClassLogger();
+
+        static Helpers()
+        {
+        }
+
+        internal static void DirectoryCopy(string sourceDirName, string destDirName, bool copySubDirs = true)
+        {
+            // Get the subdirectories for the specified directory.
+            DirectoryInfo dir = new DirectoryInfo(sourceDirName);
+
+            if (!dir.Exists)
+            {
+                throw new DirectoryNotFoundException(
+                    "Source directory does not exist or could not be found: "
+                    + sourceDirName);
+            }
+
+            DirectoryInfo[] dirs = dir.GetDirectories();
+
+            // If the destination directory doesn't exist, create it.
+            Directory.CreateDirectory(destDirName);
+
+            // Get the files in the directory and copy them to the new location.
+            FileInfo[] files = dir.GetFiles();
+            foreach (FileInfo file in files)
+            {
+                string tempPath = Path.Combine(destDirName, file.Name);
+                file.CopyTo(tempPath, false);
+            }
+
+            // If copying subdirectories, copy them and their contents to new location.
+            if (copySubDirs)
+            {
+                foreach (DirectoryInfo subdir in dirs)
+                {
+                    string tempPath = Path.Combine(destDirName, subdir.Name);
+                    DirectoryCopy(subdir.FullName, tempPath, copySubDirs);
+                }
+            }
+        }
+
+        /// <summary>
+        /// Identifies all elements in leftDirectory that either don't exist in rightDirectory or
+        /// exist with different content. This function is "smart" in that it is resilient to
+        /// changes in directory names, meaning: leftDirectory, file = /foo/bar/quux/baz.txt
+        /// rightDirectory, file = /bing/quux/baz.txt These would be correctly classified as the
+        /// same file. If there was another file in rightDirectory: rightDirectory, file =
+        /// /qwerty/bing/quux/baz.txt Then that one would match better, since it has a longer suffix
+        /// in common with the leftDirectory file.
+        /// </summary>
+        /// <param name="leftDirectory">Typically the existing package</param>
+        /// <param name="rightDirectory">Source repo, or built package, etc.</param>
+        /// <returns></returns>
+        public static IEnumerable<DirectoryDifference> DirectoryDifference(string leftDirectory, string rightDirectory, DiffTechnique diffTechnique)
+        {
+            var results = new List<DirectoryDifference>();
+
+            // left = built package, right = source repo
+            var leftFiles = Directory.EnumerateFiles(leftDirectory, "*", SearchOption.AllDirectories);
+            var rightFiles = Directory.EnumerateFiles(rightDirectory, "*", SearchOption.AllDirectories);
```

Comment:
Correct, it's a set difference (A - B), since we don't care about additional files in the source repo (right) that aren't present in the package, just the other way around.